buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/GitHubApi.kt

```diff
@@ -0,0 +1,76 @@
+package org.sonarsource.kotlin.buildsrc.tasks
+
+import com.github.kittinunf.fuel.core.FuelError
+import com.github.kittinunf.fuel.httpGet
+import com.github.kittinunf.fuel.core.Request
+import com.github.kittinunf.fuel.core.Response
+import com.github.kittinunf.fuel.gson.responseObject
+import com.github.kittinunf.result.Result
+
+private const val GITHUB_AUTH_TOKEN_ENV_NAME = "GH_API_TOKEN"
+private val GH_TOKEN by lazy {
+    System.getenv(GITHUB_AUTH_TOKEN_ENV_NAME).let {
+        if (it.isNullOrBlank()) null else it
+    }
+}
+
+private const val PR_QUERY_STRING = "is:pr is:open repo:SonarSource/rspec label:kotlin"
+
+internal object GitHubApi {
+    fun getBranchCandidatesForRule(ruleKey: String) =
+        preparePrSearchQuery(ruleKey).getResponseObject<PrSearchResults>().let { (response, result) ->
+            val (searchResults, error) = result
+            error?.let {
+                System.err.println("Request: ${response.url}")
+                System.err.println("Response: ${response.data.decodeToString()}")
+                throw it
+            }
+            searchResults?.items ?: throw IllegalStateException("Got null response while searching PRs for rule key $ruleKey.")
+        }.map { searchResult ->
+            preparePrDetailsQuery(searchResult.number).getResponseObject<Pr>().let { (response, result) ->
+                val (pr, error) = result
+                error?.let {
+                    System.err.println("Request: ${response.url}")
+                    System.err.println("Response: ${response.data.decodeToString()}")
+                    throw it
+                }
```

Comment:
maybe this logic is worth being centralized somewhere as it's duplicating lines 23-26